file/src/main/scala/akka/stream/alpakka/file/javadsl/LogRotatorSink.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.file.javadsl
+
+import java.nio.file.{Path, StandardOpenOption}
+import java.util
+import java.util.concurrent.CompletionStage
+
+import akka.Done
+import akka.stream.javadsl
+import akka.stream.javadsl.Sink
+import akka.util.ByteString
+
+import scala.collection.JavaConverters._
+
+object LogRotatorSink {
+  def createFromFunction[T](
```

Comment:
For name `create` is good for both of them and is how the Java-DSL factories most often looks.

For the parameters, there is `java.util.function.Supplier` for the factory function and `java.util.function.Function` for the actual bytes=>path function, instead of `Option` use `Optional`. This will force you to do some toJava/asJava conversions and explicit calls to `apply` of the Java types but should not be too invasive.